packages/task/src/browser/task-service.ts

```diff
@@ -970,8 +970,8 @@ export class TaskService implements TaskConfigurationClient {
             selectedRange = Range.create(startLine, 0, endLine + 1, 0);
         }
         const selectedText: string = this.editorManager.currentEditor.editor.document.getText(selectedRange).trimRight() + '\n';
-        let terminal = this.terminalService.currentTerminal;
-        if (!terminal) {
+        let terminal = this.terminalService.lastUsedTerminal;
+        if (!terminal || terminal.kind !== 'user' || (await terminal.hasChildProcesses())) {
```

Comment:
What the last predicate for? `terminal.hasChildProcesses`?

Reply:
`terminal.hasChildProcesses()` was introduced in https://github.com/eclipse-theia/theia/pull/6769, where it was used to check if the terminal that has a shell is in use.

Reply:
ok, not sure whether a user expects it, but let's try like that